### PR TITLE
API: Add Overload Manager ResetReason and ResponseFlag.

### PR DESCRIPTION
**Pull request by KBaichoo**
Signed-off-by: Kevin Baichoo <kbaichoo@google.com>

Commit Message:  Add Overload Manager ResetReason and ResponseFlag.
Additional Description: For the changes needed in Issue #15791 we will have overload manager reset streams. Given that, we'd like to have a response flag for overload manager resets. This change defines that response flag.
Risk Level: Low
Testing:Unit Tests
Docs Changes: Included
Release Notes: Included
Platform Specific Features: N/A
Related Issue #15791

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -78,6 +78,7 @@ Removed Config or Runtime
 New Features
 ------------
 
+* access_log: added the new response flag `OM` for Overload Manager termination. The error flag will be set when the http stream is terminated by overload manager.
```
Comment (fuqianggao) [inline docs/root/version_history/current.rst:81]: maybe add ref: to the proto, similar to the one below?

**Comment by antoniovicente** [comment]
You have test failures which seem related to this PR.

/wait

```
2021-06-18T18:30:14.7762567Z ==================== Test output for //test/extensions/access_loggers/grpc:grpc_access_log_utils_test:
2021-06-18T18:30:14.7763192Z [==========] Running 1 test from 1 test suite.
2021-06-18T18:30:14.7763881Z [----------] Global test environment set-up.
2021-06-18T18:30:14.7764642Z [----------] 1 test from UtilityResponseFlagsToAccessLogResponseFlagsTest
2021-06-18T18:30:14.7765238Z [ RUN      ] UtilityResponseFlagsToAccessLogResponseFlagsTest.All
2021-06-18T18:30:14.7766559Z TestRandomGenerator running with seed -1346820440
2021-06-18T18:30:14.7767412Z test/extensions/access_loggers/grpc/grpc_access_log_utils_test.cc:51: Failure
2021-06-18T18:30:14.7768045Z Expected equality of these values:
2021-06-18T18:30:14.7768573Z   common_access_log_expected.DebugString()
2021-06-18T18:30:14.7770982Z     Which is: "response_flags {\n  failed_local_healthcheck: true\n  no_healthy_upstream: true\n  upstream_request_timeout: true\n  local_reset: true\n  upstream_remote_reset: true\n  upstream_connection_failure: true\n  upstream_connection_termination: true\n  upstream_overflow: true\n  no_route_found: true\n  delay_injected: true\n  fault_injected: true\n  rate_limited: true\n  unauthorized_details {\n    reason: EXTERNAL_SERVICE\n  }\n  rate_limit_service_error: true\n  downstream_connection_termination: true\n  upstream_retry_limit_exceeded: true\n  stream_idle_timeout: true\n  invalid_envoy_request_headers: true\n  downstream_protocol_error: true\n  upstream_max_stream_duration_reached: true\n  response_from_cache_filter: true\n  no_filter_config_found: true\n  duration_timeout: true\n  upstream_protocol_error: true\n  no_cluster_found: true\n}\n"
2021-06-18T18:30:14.7773421Z   common_access_log.DebugString()
2021-06-18T18:30:14.7775880Z     Which is: "response_flags {\n  failed_local_healthcheck: true\n  no_healthy_upstream: true\n  upstream_request_timeout: true\n  local_reset: true\n  upstream_remote_reset: true\n  upstream_connection_failure: true\n  upstream_connection_termination: true\n  upstream_overflow: true\n  no_route_found: true\n  delay_injected: true\n  fault_injected: true\n  rate_limited: true\n  unauthorized_details {\n    reason: EXTERNAL_SERVICE\n  }\n  rate_limit_service_error: true\n  downstream_connection_termination: true\n  upstream_retry_limit_exceeded: true\n  stream_idle_timeout: true\n  invalid_envoy_request_headers: true\n  downstream_protocol_error: true\n  upstream_max_stream_duration_reached: true\n  response_from_cache_filter: true\n  no_filter_config_found: true\n  duration_timeout: true\n  upstream_protocol_error: true\n  no_cluster_found: true\n  overload_manager: true\n}\n"
2021-06-18T18:30:14.7778378Z With diff:
2021-06-18T18:30:14.7778774Z @@ +27,4 @@
2021-06-18T18:30:14.7779207Z    upstream_protocol_error: true
2021-06-18T18:30:14.7779671Z    no_cluster_found: true
2021-06-18T18:30:14.7780123Z +  overload_manager: true
2021-06-18T18:30:14.7780509Z  }\n
```

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (KBaichoo) [reply docs/root/version_history/current.rst]: Done.

**Review by htuch — COMMENTED** [review]
/lgtm api

**Review by antoniovicente — APPROVED** [review]
Thanks.

**Pull request merged**